analyzer/src/main/kotlin/managers/Pub.kt

```diff
@@ -75,9 +75,9 @@ private const val GRADLE_VERSION = "5.6.4"
 private const val PUB_LOCK_FILE = "pubspec.lock"
 
 private val flutterCommand = if (Os.isWindows) "flutter.bat" else "flutter"
-private val pubCommand = if (Os.isWindows) "pub.bat" else "pub"
+private val pubCommand = if (Os.isWindows) "pub.bat" else "dart pub"
```

Comment:
This is not strictly necessary and would maybe break backwards compatibility.
When downloading the flutter SDK from [flutter.dev](https://flutter.dev/docs/development/tools/sdk/releases) the `pub` binary is included ([it is deprecated](https://dart.dev/tools/pub/cmd)).

The issue is that (some?) package managers do not include it anymore (incl. Homebrew and Macports on MacOS).
Which is how most developers would install flutter on their environment.

Would it be possible to add an additional check for the below configured `flutterVersion` and adjust the command accordingly?
`dart pub` is available from dart version 2.10 ([source](https://dart.dev/tools/pub/cmd)) which is included in flutter version >= 1.22.0.

Reply:
> When downloading the flutter SDK from flutter.dev the pub binary is included (it is deprecated).

I just rechecked. This is incorrect. The SDK doesn't package the `pub` binary anymore.

Unfortunately this change doesn't work as at some point ORT tries to run `dart pub --version` which is not a valid command.